queries/usd/highlights.scm

```diff
@@ -0,0 +1,228 @@
+(None) @constant.builtin
+(asset_path) @string.special.asset
+(bool) @boolean
+(comment) @comment @spell
+(custom) @function.builtin
+(float) @float
+(integer) @number
+(namespace_identifier) @namespace
+(orderer) @function.call
+(prim_path) @string.special.prim
+(uniform) @function.builtin
+
+[
+  "class"
+  "def"
+  "over"
+] @keyword.function
```

Comment:
Class is a function? :laughing: same question for over

Reply:
It's a bit technical but I'll try to summarize. USD is a scene description format with a C++ and Python API, not a programming language itself, so "class", "def", and "over" don't refer to anything we might expect in programming terms even if you see a familiar word like "class".

If you're curious about a more technical explanation ...

`def` means "a concrete representation of data"
`over` means "something that (ideally) overrides another `def` which could be in the current file or referenced from some other file"
`class` means "a totally abstract piece of data that doesn't exist itself (like a `def`) but could be inherited by a `def`"

For an even greater technical breakdown, Pixar has a glossary on these terms which should be helpful for clarification

https://openusd.org/release/glossary.html#def

https://openusd.org/release/glossary.html#class

https://openusd.org/release/glossary.html#usdglossary-over

Please let me know if you have follow-up questions and I'll try to answer the best I can :)